rust/core-lib/src/event_context.rs

```diff
@@ -55,7 +55,7 @@ pub const MAX_SIZE_LIMIT: usize = 1024 * 1024;
 //want to make this tuneable at runtime, or to be configured by the client.
 /// The render delay after an edit occurs; plugin updates received in this
 /// window will be sent to the view along with the edit.
-const RENDER_DELAY: Duration = Duration::from_millis(2);
```

Comment:
My memory is not super-clear on this, I probably went back and forth. Some of my inspiration was the Android [INCLUSIVE/EXCLUSIVE](https://stackoverflow.com/questions/9879233/explain-the-meaning-of-span-flags-like-span-exclusive-exclusive) stuff, in which INCLUSIVE_INCLUSIVE would have meaning as zero length but the others should be removed.

If annotations have an explicit definition that includes zero-length (and it looks like they do), then I think it does make sense to include them.

Reply:
Yeah, for the plugin I was writing, often annotations were not correctly rendered after an edit. After increasing the delay the problem was resolved.

Reply:
Not sure if I can just increase it?

Reply:
It's reasonable in many cases that annotations not show up immediately (for instance with git status, or compiler warnings). In other cases it might be that the plugin is slow to return annotations? Are you doing a release build?